### PR TITLE
#14 - User input to first Tasty API 

### PR DESCRIPTION
**Pull request by EmH93**
This segment of JS includes the request for the first Tasty API based on the users input, which returns eight recipe ID numbers that we can put into the second Tasty API for the recipe cards.

### CODE REVIEW

File: js/actions.js [hunk 1]
```diff
@@ -1,7 +1,29 @@
...
\ No newline at end of file
+//On click function for search button
+searchButton.on("click", function(event){
+    event.preventDefault();
+    var userInput = searchInput.val().trim();
+//Tasty API Query one - list of meals
+    const settings = {
+        "async": true,
+        "crossDomain": true,
+        "url": "https://tasty.p.rapidapi.com/recipes/list?from=0&size=8&q=Chicken",
```
Comment (NukeTurtle) [inline js/actions.js]: Here the query is always going to search for "Chicken".  You probably want to feed it the variable "userInput" instead.

Comment (NukeTurtle) [reply js/actions.js]: Also, "size=8" is probably not what we want to search, but rather "size=1" and then limit the result of its "results" to 8.

Comment (EmH93) [reply js/actions.js]: I believe size is the number of results (see screenshot below)
image

**Review by NukeTurtle — CHANGES_REQUESTED** [review]
See comments

File: js/actions.js (hunk 1 above)
Comment (EmH93) [reply js/actions.js]: @NukeTurtle I've amended the chicken error, can you review again please :)

Comment (NukeTurtle) [reply js/actions.js]: It is, but if you go into each element of this response, you will notice number 2-8 aren't actually results that are useful to us.

Comment (EmH93) [reply js/actions.js]: This is the API that just pulls the ID number, not the API that gives us recipie information, so only the ID number is a valid output, which is then fed into the second tasty API, see below:
image

The second image below is one of those id's put into the second API
image

File: js/actions.js [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
\ No newline at end of file
+//On click function for search button
+searchButton.on("click", function(event){
+    event.preventDefault();
+    var userInput = searchInput.val().trim();
+//Tasty API Query one - list of meals
+    const settings = {
+        "async": true,
+        "crossDomain": true,
+        "url": "https://tasty.p.rapidapi.com/recipes/list?from=0&size=8&q=" + userInput,
```
Comment (NukeTurtle) [inline js/actions.js:14]: This is ok, but for reference we could also write it this way:
"url": `https://tasty.p.rapidapi.com/recipes/list?from=0&size=8&q=${userInput}`,
It looks a bit cleaner that way and you save on extra spaces etc.

Comment (NukeTurtle) [reply js/actions.js:14]: Re "size" property, have a look at this:
image
then this:
image

The latter will still return all (28) recipes, but now you don't have extra calls to the API and overhead of consecutive levels within result.

**Review by NukeTurtle — APPROVED** [review]

**Pull request merged**